pysmt/fnode.py

```diff
@@ -854,6 +861,11 @@ def __rshift__(self, right):
 
     def __mod__(self, right):
         return self._apply_infix(right, None, bv_function=_mgr().BVURem)
+
+    def __call__(self, *args):
```

Comment:
If args is empty ``f()`` the iterator will not execute and neither  ``_infix_prepare_args``. 

Also, don't you should not check against the type of the function, but the type of the argument. See comment below.

Reply:
Done.